
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>DTRAPipeline.utils package &#8212; DTRAPipe 1.7.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="DTRAPipeline.databases package" href="DTRAPipeline.databases.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="dtrapipeline-utils-package">
<h1>DTRAPipeline.utils package<a class="headerlink" href="#dtrapipeline-utils-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-DTRAPipeline.utils.SILVA16S">
<span id="dtrapipeline-utils-silva16s-module"></span><h2>DTRAPipeline.utils.SILVA16S module<a class="headerlink" href="#module-DTRAPipeline.utils.SILVA16S" title="Permalink to this headline">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.SILVA16S.blast_guides_outgroup">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.SILVA16S.</span></span><span class="sig-name descname"><span class="pre">blast_guides_outgroup</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">self</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ingroup_fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outgroup_fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">perc_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qcov_hsp_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold_num</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">total_records</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7000000.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Guides'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.SILVA16S.blast_guides_outgroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Blast against outgroups to remove less specific fasta entries</p>
<p>Input:</p>
<p>ingroup_fasta_path: str, path to ingroup fasta
outgroup_fasta_path: str, path to outgroup fasta
perc_identity: float, exact match threshold accepted
qcov_hsp_perc: int, query coverage length threshold
word_size: size to use in blast query, larger word_sizes are much faster
threshold_num: int, number of unique outgroup hits to allow
total_records: float, number of outgroup records
plot: bool, plot the blast output
title: str, title for plot</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.blastutils">
<span id="dtrapipeline-utils-blastutils-module"></span><h2>DTRAPipeline.utils.blastutils module<a class="headerlink" href="#module-DTRAPipeline.utils.blastutils" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.blastutils.</span></span><span class="sig-name descname"><span class="pre">BlastUtils</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">preset</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><a class="reference internal" href="#DTRAPipeline.utils.presets.BaseType" title="DTRAPipeline.utils.presets.BaseType"><span class="pre">DTRAPipeline.utils.presets.BaseType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.add_missing_hits">
<span class="sig-name descname"><span class="pre">add_missing_hits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_bout</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_fasta</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.add_missing_hits" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_outgroup">
<span class="sig-name descname"><span class="pre">blast_guides_outgroup</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ingroup_fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outgroup_fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_tab_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">perc_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qcov_hsp_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold_num</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">total_records</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5000000.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Guides'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_target_seqs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_outgroup" title="Permalink to this definition">¶</a></dt>
<dd><p>Blast against outgroups to remove less specific fasta entries</p>
<p>Input:</p>
<p>ingroup_fasta_path: str, path to ingroup fasta
outgroup_fasta_path: str, path to outgroup fasta
perc_identity: float, exact match threshold accepted
qcov_hsp_perc: int, query coverage length threshold
word_size: size to use in blast query, larger word_sizes are much faster
threshold_num: int, number of unique outgroup hits to allow
total_records: float, number of outgroup records
plot: bool, plot the blast output
title: str, title for plot</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_sensitivity">
<span class="sig-name descname"><span class="pre">blast_guides_sensitivity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guides_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fasta_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_qcov_sensitivity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_piden_sensitivity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_tab_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">write</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_adjusted_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_subjects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Guides'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_sensitivity" title="Permalink to this definition">¶</a></dt>
<dd><p>function that blasts the universal guides, or a given path against either the outlier fasta
if it exists or the curated fasta.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_specificity">
<span class="sig-name descname"><span class="pre">blast_guides_specificity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guides_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative_taxids</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_adjusted_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_qcov_specificity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_piden_specificity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_fasta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">15</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_target_seqs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.blast_guides_specificity" title="Permalink to this definition">¶</a></dt>
<dd><p>Function that blasts guides, rois, probes against the entire nt database with the exception
of the negative_taxids included as well as all synthetic taxids. It will subsequently
rewrite all probes that do not have hits to any species.</p>
<p>guides_path: str, to probes or existing universal guides.
negative taxids: list, of int taxids to be appended to the synthetic taxids
db: str, path to nt database, or other database of choosing
plot: bool, plots output</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.blast_tax_checker">
<span class="sig-name descname"><span class="pre">blast_tax_checker</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.blast_tax_checker" title="Permalink to this definition">¶</a></dt>
<dd><p>Check for the taxonomy files needed to run the specificity blast, if they don’t exist fetch and download them.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.get_fda_file">
<span class="sig-name descname"><span class="pre">get_fda_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">negative_taxids</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.get_fda_file" title="Permalink to this definition">¶</a></dt>
<dd><p>function which retrieves the fda leaf taxids and synthetic taxids and subtracts the
synthetic and target taxid_list be used in the specificity analysis.</p>
<p>negative_taxids: list, taxids that should not be queried for a specificity analysis</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.get_species_taxids">
<span class="sig-name descname"><span class="pre">get_species_taxids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.get_species_taxids" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve all descendant taxids using the ncbi get_species_taxids script.
Parameters:</p>
<p>taxids: list, list of ncbi taxids</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.get_synthetic_list">
<span class="sig-name descname"><span class="pre">get_synthetic_list</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">negative_taxids</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.get_synthetic_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves the synthetic taxids and adds the target id to the list then writes a local copy
for the blast negative_taxidlist commandline parametr.</p>
<p>negative_taxids: list, of taxids that should not be considered for specificity analysis.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.process_blast">
<span class="sig-name descname"><span class="pre">process_blast</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">option</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'sensitivity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_tab_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_fasta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_fasta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">write</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Guides'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_adjusted_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_subjects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.process_blast" title="Permalink to this definition">¶</a></dt>
<dd><p>Analyzes the blast output and can remove less sensitive hits in a
regenerated fasta file. Alternatively,  Can handle specificity and sensitivity analysis.</p>
<p>option: str, sensitivity or specificity
blast_tab_path: str, path to blast_tab, usually should not be used.
plot: bool, primarily used for sensitivity analysis</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.process_sensitive_blast">
<span class="sig-name descname"><span class="pre">process_sensitive_blast</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">blast_tab_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_subjects</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.process_sensitive_blast" title="Permalink to this definition">¶</a></dt>
<dd><p>Formats and calculates the results from a blast sensitiity subject query</p>
<p>blast_tab_path: str/stringIO, path to blast tab file or buffered blast results.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.read_fmt6_blast">
<span class="sig-name descname"><span class="pre">read_fmt6_blast</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.read_fmt6_blast" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the format 6 blast tab file output</p>
<p>Input:
path: str, blast tab path
file_type: str, blast paradigm (None, ‘specificity’)</p>
<p>Output:
df_bout2: pd.DataFrame, loaded dataframe</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.write_sensitive_guides">
<span class="sig-name descname"><span class="pre">write_sensitive_guides</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_bout</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">option</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_fasta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_adjusted_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.write_sensitive_guides" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes sensitivity options while accomodating the outputs. Sensitive guides
that have an adjusted blast_adjusted_perc below the desired range will not be written to the updated
guides file.</p>
<p>Input:</p>
<p>df_bout: pandas dataframe, tab file output from blast
option: str, sensitivity or specificity
output_fasta: str, name of output fasta
input_fasta: str, path to read guides from
blast_adjusted_perc: float, threshold with hich the adjusted percentage must exceed.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.blastutils.BlastUtils.write_specific_guides">
<span class="sig-name descname"><span class="pre">write_specific_guides</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_bout</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">option</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_fasta</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_adjusted_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.blastutils.BlastUtils.write_specific_guides" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes specificity guides that have all blast hits subtracted while accomodating the two outputs.</p>
<p>Input:</p>
<p>df_bout: pandas dataframe, tab file output from blast
option: str, sensitivity or specificity
output_fasta: str, name of output fasta
input_fasta: str, path to read guides from
blast_adjusted_perc: float, threshold with hich the adjusted percentage must exceed.</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.datacleaner_updated">
<span id="dtrapipeline-utils-datacleaner-updated-module"></span><h2>DTRAPipeline.utils.datacleaner_updated module<a class="headerlink" href="#module-DTRAPipeline.utils.datacleaner_updated" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Alphabet">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">Alphabet</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Alphabet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Alphabet.all_letters">
<span class="sig-name descname"><span class="pre">all_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Alphabet.all_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Alphabet.extended_letters">
<span class="sig-name descname"><span class="pre">extended_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Alphabet.extended_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Alphabet.generate_random_sequence">
<span class="sig-name descname"><span class="pre">generate_random_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Alphabet.generate_random_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a random sequence drawn from an alphabet.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><p>length (int): length of random sequence to be generated
sequence_type (chr): one of the allowed ALPHABETS (nucleic_acid, protein, glycan)</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Alphabet.standard_letters">
<span class="sig-name descname"><span class="pre">standard_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Alphabet.standard_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">DataCleaner</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq_column</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_column</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">classify</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'median'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">augmentation_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'none'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.binned_target_values">
<span class="sig-name descname"><span class="pre">binned_target_values</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.binned_target_values" title="Permalink to this definition">¶</a></dt>
<dd><p>Return binned and preprocessed target values</p>
<p>Bins continuous target data to get labels for classification using the
user-defined threshold, if provided, or defaults to using the median value.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.clean">
<span class="sig-name descname"><span class="pre">clean</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.clean" title="Permalink to this definition">¶</a></dt>
<dd><p>Read and clean the input data.</p>
<p>Read the input data, process the column containing sequences, and process
the column containing the targets</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.consensus_sequence">
<span class="sig-name descname"><span class="pre">consensus_sequence</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.consensus_sequence" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.letter_frequency_matrix">
<span class="sig-name descname"><span class="pre">letter_frequency_matrix</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.letter_frequency_matrix" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_autobinning_metrics">
<span class="sig-name descname"><span class="pre">plot_autobinning_metrics</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_autobinning_metrics" title="Permalink to this definition">¶</a></dt>
<dd><p>If continuous target is automatically binned, this function plots the p-values.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_character_heatmap">
<span class="sig-name descname"><span class="pre">plot_character_heatmap</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_character_heatmap" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot character frequencies as a function of sequence position</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_target_distributions">
<span class="sig-name descname"><span class="pre">plot_target_distributions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.plot_target_distributions" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot cleaned target values as well as the original</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.DataCleaner.preprocessed_target_values">
<span class="sig-name descname"><span class="pre">preprocessed_target_values</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.DataCleaner.preprocessed_target_values" title="Permalink to this definition">¶</a></dt>
<dd><p>Return preprocessed target values</p>
</dd></dl>

</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Error">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">Error</span></span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Error" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>Base class for exceptions in this module.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.InputError">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">InputError</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">message</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.InputError" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.datacleaner_updated.Error" title="DTRAPipeline.utils.datacleaner_updated.Error"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.datacleaner_updated.Error</span></code></a></p>
<p>Exception raised for errors in the input.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><p>expression: input expression in which the error occurred
message: explanation of the error</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">Sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence.augmented">
<span class="sig-name descname"><span class="pre">augmented</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">augmentation_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence.augmented" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the augmented (complemented/reverse complemented) sequence. Applicable for nucleic_acid only!</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence.fill">
<span class="sig-name descname"><span class="pre">fill</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence.fill" title="Permalink to this definition">¶</a></dt>
<dd><p>Replace gaps (“-“) with the corresponding gap letter.
Not done for glycans.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence.pad">
<span class="sig-name descname"><span class="pre">pad</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence.pad" title="Permalink to this definition">¶</a></dt>
<dd><p>Pad sequence to desired length using the appropriate gap letter</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence.scrambled">
<span class="sig-name descname"><span class="pre">scrambled</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence.scrambled" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the scrambled sequence</p>
<p>Ensures that the scrambled output returned has the same letter frequency
but in a different order than the original sequence. Negative control.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.Sequence.standardized">
<span class="sig-name descname"><span class="pre">standardized</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.Sequence.standardized" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the fully standardized sequence</p>
<p>Fill gaps, add padding, map non-standard letters and ensure all letters are valid</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.compute_tests">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">compute_tests</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_eval</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.compute_tests" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes the result of the data splitting as a function of various percentiles and computes
significance values between the test and control sets.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.evaluate_partitions">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">evaluate_partitions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">keep_bin_edges</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_processed</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.evaluate_partitions" title="Permalink to this definition">¶</a></dt>
<dd><p>This function evaluates a lightweight classifier according to the thresholds.
Inputs are a list of bin-edges for the continuous target and the processed df.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.partition_data">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">partition_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_processed</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.partition_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Partitions the processed dataframe into two classes provided that the target is continuous.
The input must already be cleaned using the data-cleaner.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.process_glycans">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">process_glycans</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">glycan_list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.process_glycans" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper function to process list of glycans into glycoletters.
Input is a list of glycans. Output is a list of glycowords</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.read_data">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">read_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.read_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Read input data in csv, .xls, or .xlxs format</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.repeat_auto_partitioning">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">repeat_auto_partitioning</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">keep_bin_edges</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_processed</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.repeat_auto_partitioning" title="Permalink to this definition">¶</a></dt>
<dd><p>Repeats the training of a Naive-Bayes Classifier on the Sequence according to a preset threshold. 
Returns the results of all calculations as well as the averaged df with a suggestion for the auto-binning.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.small_glymotif_find">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">small_glymotif_find</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">s</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.small_glymotif_find" title="Permalink to this definition">¶</a></dt>
<dd><p>Breaks down glycans into list of monomers. Assumes
that user provides glycans in a consistent universal 
notation. Input is a glycan string</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.datacleaner_updated.split_transform_data">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.datacleaner_updated.</span></span><span class="sig-name descname"><span class="pre">split_transform_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">X</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">X_control</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.datacleaner_updated.split_transform_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Splits the processed data into training, testing, and controls.</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.entrezutils">
<span id="dtrapipeline-utils-entrezutils-module"></span><h2>DTRAPipeline.utils.entrezutils module<a class="headerlink" href="#module-DTRAPipeline.utils.entrezutils" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.entrezutils.</span></span><span class="sig-name descname"><span class="pre">AutoEntrez</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">email</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncbi_api_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>The AutoEntrez object automatically selects and retrieves the representatives and all records
as they are available within NCBI. The object additionally will retrieve customly set parameters.</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.check_genome_params">
<span class="sig-name descname"><span class="pre">check_genome_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.check_genome_params" title="Permalink to this definition">¶</a></dt>
<dd><p>Search for custom parameters if they exist and retrieve associated records.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.check_rep_params">
<span class="sig-name descname"><span class="pre">check_rep_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.check_rep_params" title="Permalink to this definition">¶</a></dt>
<dd><p>Search for custom representative parameters if they exist and retrieve associated records.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.download_genomes">
<span class="sig-name descname"><span class="pre">download_genomes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.download_genomes" title="Permalink to this definition">¶</a></dt>
<dd><p>Search for genomes if custom parameters are not set otherwise retrieve custom parameters.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.get_params">
<span class="sig-name descname"><span class="pre">get_params</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.get_params" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.select_rep">
<span class="sig-name descname"><span class="pre">select_rep</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.select_rep" title="Permalink to this definition">¶</a></dt>
<dd><p>Search for representatives if custom paramaters are not set.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.AutoEntrez.set_params">
<span class="sig-name descname"><span class="pre">set_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.AutoEntrez.set_params" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.entrezutils.get_lineage_df">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.entrezutils.</span></span><span class="sig-name descname"><span class="pre">get_lineage_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">NCBI_pathogen_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.entrezutils.get_lineage_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Function that retrieves the full lineage for a particular NCBI_pathogen_name</p>
<p>Input:
NCBI_pathogen_name: str, ncbi taxonomy associated name</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.features">
<span id="dtrapipeline-utils-features-module"></span><h2>DTRAPipeline.utils.features module<a class="headerlink" href="#module-DTRAPipeline.utils.features" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.features.FeatureFinder">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.features.</span></span><span class="sig-name descname"><span class="pre">FeatureFinder</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">preset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.features.FeatureFinder" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.features.FeatureFinder.extract_cds">
<span class="sig-name descname"><span class="pre">extract_cds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gb_feature</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gb_recs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.features.FeatureFinder.extract_cds" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.features.FeatureFinder.get_features">
<span class="sig-name descname"><span class="pre">get_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">acc_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_duplicates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.features.FeatureFinder.get_features" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.features.FeatureFinder.get_gbff_features">
<span class="sig-name descname"><span class="pre">get_gbff_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gbff_handle</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">drop_duplicates</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.features.FeatureFinder.get_gbff_features" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.info_utils">
<span id="dtrapipeline-utils-info-utils-module"></span><h2>DTRAPipeline.utils.info_utils module<a class="headerlink" href="#module-DTRAPipeline.utils.info_utils" title="Permalink to this headline">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.align_unique_refs">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">align_unique_refs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_r2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.align_unique_refs" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes a multi-record fasta file of unique sequences and generates a MAFFT 
alignment</p>
<p>Inputs:
df_r2: cleaned_df of genomic sequences
base_file_name: str - name of the pathogen
out_dir: str - path to dir where alignment should be stored</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.annotate_contig">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">annotate_contig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.annotate_contig" title="Permalink to this definition">¶</a></dt>
<dd><p>This function locates a contig within a reference genome and assigns an indicator
depending on how many instances that contig is found within the reference genome.</p>
<p>Inputs:
seq: str - nucleic acid
ref_genome: str - nucleic acid</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.calc_mash_distance">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">calc_mash_distance</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subject_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">collapse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.calc_mash_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses MASH to calculate the pw distance between a reference hash table and 
a subject list of genomes.</p>
<dl class="simple">
<dt>Inputs:</dt><dd><p>Note: Use make_mash_input_file to make the .txt files</p>
</dd>
</dl>
<p>out_dir: str - path to the directory where files should be stored
reference_path: str - path to a .txt file listing the reference group of genomes
subject_path: str - path to a .txt file listing the subject group of genomes
base_file_name: str - name of the pathogen
collapse: bool - if True, collapses the in-group onto one genome</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.calc_plot_pwsim">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">calc_plot_pwsim</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_ids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_samples</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.calc_plot_pwsim" title="Permalink to this definition">¶</a></dt>
<dd><p>This function calculates and plots a sampling of pairwise similarity heatmap.
A hierarchical clustering is subsequently implemented in order to cluster the sequences.</p>
<p>Inputs: 
aln_arr: numpy array of alignment 
aln_ids: list of corresponding fasta IDs
n_samples: int - number of sequence samples to draw</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.calc_protein_pwsim">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">calc_protein_pwsim</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_ids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.calc_protein_pwsim" title="Permalink to this definition">¶</a></dt>
<dd><p>This function calculates and plots a sampling of pairwise similarity heatmap.
Also clusters by distance. Keep shape of array to less than 50</p>
<p>Inputs: 
aln_arr: numpy array of alignment 
aln_ids: list of corresponding fasta IDs</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.cluster_seqs">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">cluster_seqs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_r2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_clusters</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.cluster_seqs" title="Permalink to this definition">¶</a></dt>
<dd><p>This function sweeps through a range of radii to cluster sequences. Returns a memory error if sequences are larger than 100kb</p>
<p>Inputs:
df_r2: thresholded ref_seq df 
base_file_name: str which is the name of the output file
out_dir: str - path to directory where records are kept
plot_clusters: Boolean</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_clusters">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_clusters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">iden_frac</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pathogen_seqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the UCLUST routines to cluster sequences 
iden_frac: float between [0,1]
pathogen_seqs: multi-record fasta file containing all the genome sequences to be clustered
out_name: str which is the name of the output file</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_contig_coords">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_contig_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cleaned_contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_contig_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>This function identifies the positions of the cleaned contigs relative to a reference genome.</p>
<p>Inputs:
cleaned_contigs: str - list of contigs as identified by skif the meet the minimum criteria</p>
<p>ref_genome: str - sequence of the ref genome that contigs belong to</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_coverage">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regions</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_coverage" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the coverage of the identified regions relative to a reference genome.</p>
<p>Inputs:
regions: list of str
ref_genome: str</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_intercontig_gap">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_intercontig_gap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_contigs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_intercontig_gap" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the gap between contigs with respect to the ref_genome
For N contigs, there are N - 1 gaps</p>
<p>Inputs:
df_contig - df - output of compute_contig_coords</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_local_unp">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_local_unp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">target_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bind_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">28</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temperature</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">37</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">struct_window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">80</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_local_unp" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the local structure of a target RNA. 
Output is the probability that a given base is unpaired.</p>
<p>Inputs:
target_name: str
target_seq: str
bind_len: int - size of probe binding region
temperature: float - temperature
struct_window: int - size of the structural window to be considered when solving the local RNA structure</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_plot_conservation">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_plot_conservation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">28</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_plot_conservation" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.compute_pw_galn_score">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">compute_pw_galn_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.compute_pw_galn_score" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the pairwise global alignment score for a probe sequence 
against the target sequence. It normalizes the score against the probe itself.
Also provides the start and end locations of the local alignment</p>
<p>Inputs:
probe - str: nucleic_acid
target - str: nucleic_acid</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.count_gaps">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">count_gaps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tile_seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.count_gaps" title="Permalink to this definition">¶</a></dt>
<dd><p>The function counts the number of gaps in a tile-sequence.</p>
<p>Inputs:
tile_seq: str</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.draw_mini_alignment">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">draw_mini_alignment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">draw_aln</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.draw_mini_alignment" title="Permalink to this definition">¶</a></dt>
<dd><p>This function draws a mini alignment graph given a MAFFT aligned MSA</p>
<p>Inputs:
aln_name: path to MAFFT MSA (fasta format)
base_file_name: str 
draw_aln: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.draw_protein_alignment">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">draw_protein_alignment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">draw_aln</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.draw_protein_alignment" title="Permalink to this definition">¶</a></dt>
<dd><p>This function draws a mini alignment graph given a MAFFT aligned MSA</p>
<p>Inputs:
aln_name: path to MAFFT MSA (fasta format)
base_file_name: str 
draw_aln: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.filter_pw_mash_distances">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">filter_pw_mash_distances</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">avoid_pattern</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.filter_pw_mash_distances" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses the distance df to obtain the nearest neighbors.</p>
<p>Inputs:
df: df computed and parsed using parse_pw_mash_distances</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.get_spacer_nbhd">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">get_spacer_nbhd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">spacer</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.get_spacer_nbhd" title="Permalink to this definition">¶</a></dt>
<dd><p>This function calculates the target_sequence nbhd for a spacer and target_seq.
Suitable for primer3 heterodimerization bindings.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.get_vienna_dg">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">get_vienna_dg</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe_seqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seqs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temperature</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.get_vienna_dg" title="Permalink to this definition">¶</a></dt>
<dd><p>This function extracts just the dG of binding using ViennaRNA cofold package.</p>
<p>Inputs:
probe_seqs: list - each element is a nucleic_acid string
target_seqs: list - each element is a nucleic_acid string
temperature: float - temperature of the binding reaction</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.join_adjacent_tiles">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">join_adjacent_tiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_sc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cons_thresh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.92</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.join_adjacent_tiles" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes the df of kmer tiles where the sensitivity was calculated</p>
<p>Inputs:
cns_seq: consensus sequence
df_sc: df that is the output of score_alignment_conservation_byposition
base_file_name: str
cons_thresh: float</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.make_mash_input_file">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">make_mash_input_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fname</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_suffix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.fna'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.make_mash_input_file" title="Permalink to this definition">¶</a></dt>
<dd><p>This function goes through a directory of genomes (fasta format) and creates a mash compatible input file.</p>
<p>Inputs
out_dir: str - path to directory containing fasta records
fname: str - name of the text file to be created</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.parse_blast_output">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">parse_blast_output</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bout</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">avoid_pattern</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.parse_blast_output" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses the blast output to remove any hits that match the pattern.</p>
<p>Inputs:
bout: df that is the output of tile_and_blast_regs
avoid_pattern: str - must be lower case</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.parse_pw_mash_distances">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">parse_pw_mash_distances</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dist_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.parse_pw_mash_distances" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses a tab-separated distance matrix as computed by mash and returns a cleaned df.</p>
<p>Inputs:
dist_path: str - path to the distance file</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.parse_smash_nn">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">parse_smash_nn</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query_col</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pattern</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_nearest</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.parse_smash_nn" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses the sourmash output from sourmash search to collapse hits to species name.
Assumption is that the first word is always the RefSeq ID followed by the species name as two separate words</p>
<p>Inputs:
df: df - contains ‘similarity’ and ‘name’
query_col: str - name of the column to be searched for the pattern
pattern: list of str - contains patterns to be searched
num_nearest: int - number of nearest neighbors to return</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.parse_txids_txt_file">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">parse_txids_txt_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">txid_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.parse_txids_txt_file" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses a txt file which is the output of gimme_taxa.py
and obtains a list of descendant taxids and names if desired as well.</p>
<p>Inputs:
txid_path: path to txt file containing a column called descendent_taxid
get_names: boolean - to also extract the scientific names</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.plot_aln_covg">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">plot_aln_covg</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_ids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">draw_logo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.plot_aln_covg" title="Permalink to this definition">¶</a></dt>
<dd><p>This function plots the coverage across all alignments.</p>
<p>Inputs:
aln_arr: numpy array of aln
aln_ids: corresponding list of fasta names
base_file_name: str
draw_logo: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.plot_dist_hm">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">plot_dist_hm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_sig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.plot_dist_hm" title="Permalink to this definition">¶</a></dt>
<dd><p>This function plots a heatmap of the pw MASH distances.
If desired, it can also plot the log p-value significance as a heatmap</p>
<p>Inputs:
df: df contained mash distances</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.plot_smoothed_aln_score">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">plot_smoothed_aln_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_sc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_len</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.plot_smoothed_aln_score" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots the smoothed alignment score as a function of window length and different windows.</p>
<p>Inputs:
cns_seq: consensus_sequence
df_sc: df - that is the output of ensemble_kmer_conservation
window_len: int</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.score_alignment_conservation_byposition">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">score_alignment_conservation_byposition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kmer_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">14</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_cons</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.score_alignment_conservation_byposition" title="Permalink to this definition">¶</a></dt>
<dd><p>This function breaks up an MSA into kmers and checks the conservation of each kmer against the whole alignment.</p>
<p>cns_seq: str - consensus sequence generated from the alignment
aln_arr: np array of alignment
gap_size: int
kmer_size: int
plot_cons: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.score_gRNA_conservation_byposition">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">score_gRNA_conservation_byposition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_grna</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_cons</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.score_gRNA_conservation_byposition" title="Permalink to this definition">¶</a></dt>
<dd><p>The function scores the conservation of each guide given an alignment.</p>
<p>cns_seq: str - consensus sequence generated from the alignment
aln_arr: np array of alignment
df_grna: df of gRNA sequences
plot_cons: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.score_tile">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">score_tile</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tile_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aligned_seqs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.score_tile" title="Permalink to this definition">¶</a></dt>
<dd><p>This function scores each tile against a set of aligned sequences.</p>
<p>Inputs:
tile_seq: str
lp: int - left position
rp: int - right position
aligned_seqs: list of str where tile_seq can be found</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.search_pattern">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">search_pattern</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query_col</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pattern</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.search_pattern" title="Permalink to this definition">¶</a></dt>
<dd><p>This function rates a df of distances for the presence of a pattern in the query description.</p>
<p>Inputs:
df: df - contains distances and p-values
query_col: str - name of the column to be searched for the pattern
pattern: list of str - contains patterns to be searched</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.smooth">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">smooth</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">28</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'hamming'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.smooth" title="Permalink to this definition">¶</a></dt>
<dd><p>smooth the data using a window with requested size.</p>
<p>This method is based on the convolution of a sour_scaled window with the signal.
The signal is prepared by introducing reflected copies of the signal 
(with the window size) in both ends so that transient parts are minimized
in the begining and end part of the output signal.</p>
<dl>
<dt>input:</dt><dd><p>x: the input signal 
window_len: the dimension of the smoothing window; should be an odd integer
window: the type of window from ‘flat’, ‘hanning’, ‘hamming’, ‘bartlett’, ‘blackman’</p>
<blockquote>
<div><p>flat window will produce a moving average smoothing.</p>
</div></blockquote>
</dd>
<dt>output:</dt><dd><p>the smoothed signal</p>
</dd>
</dl>
<p>example:</p>
<p>t=linspace(-2,2,0.1)
x=sin(t)+randn(len(t))*0.1
y=smooth(x)</p>
<p>see also:</p>
<p>numpy.hanning, numpy.hamming, numpy.bartlett, numpy.blackman, numpy.convolve
scipy.signal.lfilter</p>
<p>TODO: the window parameter could be the window itself if an array instead of a string
NOTE: length(output) != length(input), to correct this: return y[(window_len/2-1):-(window_len/2)] instead of just y.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.stitch_overlap_guides">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">stitch_overlap_guides</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_good2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spacer_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">28</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.stitch_overlap_guides" title="Permalink to this definition">¶</a></dt>
<dd><p>This function stitches overlapping guide segments into contiguous regions.
Returns a list of contiguous regions</p>
<p>Inputs:
df_good2: df contains</p>
<blockquote>
<div><p>gRNA_seq
gRNA_lp
gRNA_rp</p>
</div></blockquote>
<p>dc: str - dummy consensus
spacer_length: int - length of the spacer</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.summarize_guide_sensitivity">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">summarize_guide_sensitivity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">blast_out</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.summarize_guide_sensitivity" title="Permalink to this definition">¶</a></dt>
<dd><p>This function summarizes the sensitivity for each guide/region in a blast-output.</p>
<p>Inputs:
blast_out: str - path to the tabular blast output file</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.tile_and_blast_regs">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">tile_and_blast_regs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regs_filt</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ntaxids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">primer_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">22</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.tile_and_blast_regs" title="Permalink to this definition">¶</a></dt>
<dd><p>This function tiles the conserved regions into primer_length pieces. Any gapped kmers 
are excluded. A blastn-short search against the nt database is conducted. This is extremely time-consuming.</p>
<p>Inputs:
regs_filt: list of str - nucleic acid sequences of the conserved regions above the desired sensitivity threshold
base_file_name: str - name of the pathogen
out_dir: str - path to directory where results ought to be stored
ntaxids: str - list of negative_taxids to be excluded from the blastn search
primer_length: int - size of the blast tile (cannot exceed 50)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.info_utils.vienna_cofold">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.info_utils.</span></span><span class="sig-name descname"><span class="pre">vienna_cofold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">s1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">s2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temperature</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.info_utils.vienna_cofold" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the ViennaRNA co-fold package to calculate the folding metrics for two RNA sequences</p>
<p>Inputs:
s1: str - nucleic_acid
s2: str - nucleic_acid
temperature: float - temperature of bi-molecular reaction</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.mfunct">
<span id="dtrapipeline-utils-mfunct-module"></span><h2>DTRAPipeline.utils.mfunct module<a class="headerlink" href="#module-DTRAPipeline.utils.mfunct" title="Permalink to this headline">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.compute_intercontig_gap">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">compute_intercontig_gap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_contigs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.compute_intercontig_gap" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the gap between contigs with respect to the ref_genome
For N contigs, there are N - 1 gaps</p>
<p>Inputs:
df_contig - df - output of compute_contig_coords</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.compute_pw_galn_score">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">compute_pw_galn_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.compute_pw_galn_score" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the pairwise global alignment score for a probe sequence
against the target sequence. It normalizes the score against the probe itself.
Also provides the start and end locations of the local alignment</p>
<p>Inputs:
probe - str: nucleic_acid
target - str: nucleic_acid</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.count_gaps">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">count_gaps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tile_seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.count_gaps" title="Permalink to this definition">¶</a></dt>
<dd><p>The function counts the number of gaps in a tile-sequence.</p>
<p>Inputs:
tile_seq: str</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.fill_refseq_template">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">fill_refseq_template</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">refseq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.fill_refseq_template" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes dataframe with contig locations within the refseq and reassembles them
based on the compute_contig_coords output.</p>
<p>df_contigs: str, df_ccompute_contig_coords dataframe
refseq: str, full string format of refseq</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.join_adjacent_tiles">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">join_adjacent_tiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_sc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cons_thresh</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.92</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.join_adjacent_tiles" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes the df of kmer tiles where the sensitivity was calculated</p>
<p>Inputs:
cns_seq: consensus sequence
df_sc: df that is the output of score_alignment_conservation_byposition
base_file_name: str
cons_thresh: float</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.join_refseq_tiles">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">join_refseq_tiles</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tiles</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">refseq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.join_refseq_tiles" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes tiles and a refseq and joins them together
tiles: list, of tile records
refseq: str, reference to be aligned against</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.join_windows">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">join_windows</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gaps</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.join_windows" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.pssm_to_dict">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">pssm_to_dict</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.pssm_to_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.score_alignment_conservation_byposition">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">score_alignment_conservation_byposition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_arr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kmer_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">14</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.score_alignment_conservation_byposition" title="Permalink to this definition">¶</a></dt>
<dd><p>This function breaks up an MSA into kmers and checks the conservation of each kmer against the whole alignment.</p>
<p>cns_seq: str - consensus sequence generated from the alignment
aln_arr: np array of alignment
gap_size: int
kmer_size: int
plot_cons: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.score_alignment_pssm">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">score_alignment_pssm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">summary_align</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap_ref</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tile_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">20</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pssm_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.92</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.score_alignment_pssm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.score_tile">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">score_tile</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tile_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aligned_seqs</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.score_tile" title="Permalink to this definition">¶</a></dt>
<dd><p>This function scores each tile against a set of aligned sequences.</p>
<p>Inputs:
tile_seq: str
lp: int - left position
rp: int - right position
aligned_seqs: list of str where tile_seq can be found</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.mfunct.smooth">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.mfunct.</span></span><span class="sig-name descname"><span class="pre">smooth</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_len</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'hamming'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.mfunct.smooth" title="Permalink to this definition">¶</a></dt>
<dd><p>smooth the data using a window with requested size.</p>
<p>This method is based on the convolution of a scaled window with the signal.
The signal is prepared by introducing reflected copies of the signal
(with the window size) in both ends so that transient parts are minimized
in the begining and end part of the output signal.</p>
<dl>
<dt>input:</dt><dd><p>x: the input signal
window_len: the dimension of the smoothing window; should be an odd integer
window: the type of window from ‘flat’, ‘hanning’, ‘hamming’, ‘bartlett’, ‘blackman’</p>
<blockquote>
<div><p>flat window will produce a moving average smoothing.</p>
</div></blockquote>
</dd>
<dt>output:</dt><dd><p>the smoothed signal</p>
</dd>
</dl>
<p>example:</p>
<p>t=linspace(-2,2,0.1)
x=sin(t)+randn(len(t))*0.1
y=smooth(x)</p>
<p>see also:</p>
<p>numpy.hanning, numpy.hamming, numpy.bartlett, numpy.blackman, numpy.convolve
scipy.signal.lfilter</p>
<p>TODO: the window parameter could be the window itself if an array instead of a string
NOTE: length(output) != length(input), to correct this: return y[(window_len/2-1):-(window_len/2)] instead of just y.</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.phylogeny">
<span id="dtrapipeline-utils-phylogeny-module"></span><h2>DTRAPipeline.utils.phylogeny module<a class="headerlink" href="#module-DTRAPipeline.utils.phylogeny" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.phylogeny.</span></span><span class="sig-name descname"><span class="pre">Phylogeny</span></span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny.build_tree">
<span class="sig-name descname"><span class="pre">build_tree</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">msa_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dtype</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'identity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'fasta'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny.build_tree" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny.get_branches">
<span class="sig-name descname"><span class="pre">get_branches</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tree</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny.get_branches" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to retrieve all branches as well as remove any calculated artifical branches
for final display to</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny.get_distance">
<span class="sig-name descname"><span class="pre">get_distance</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">msa_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dtype</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'identity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aln_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'fasta'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny.get_distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to retrieve a distance matrix from an msa file. Automatically removes
and duplicates introduced in the mafft alignment.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny.get_outliers">
<span class="sig-name descname"><span class="pre">get_outliers</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">branch_df</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny.get_outliers" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.phylogeny.Phylogeny.load_tree">
<span class="sig-name descname"><span class="pre">load_tree</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tree_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tree_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'newick'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fig_scale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.phylogeny.Phylogeny.load_tree" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.plot">
<span id="dtrapipeline-utils-plot-module"></span><h2>DTRAPipeline.utils.plot module<a class="headerlink" href="#module-DTRAPipeline.utils.plot" title="Permalink to this headline">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.draw_mini_alignment">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">draw_mini_alignment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alignment_array</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alignment_ids</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.draw_mini_alignment" title="Permalink to this definition">¶</a></dt>
<dd><p>This function draws a mini alignment graph given a MAFFT aligned MSA</p>
<p>Inputs:
alignment_array: np.array
alignment_ids: np.array
title: str, plot title</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.jaccard_plot">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">jaccard_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">title</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dfc</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.jaccard_plot" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_blast_tab">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_blast_tab</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_bout</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_bout2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">option</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pssm_perc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_number</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_blast_tab" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
df_bout: pandas.DataFrame, blast output from tab file.
Plot the number and distribution of viable kmers that follow qcov and ident blast cutoffs</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_centroid_sampling">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_centroid_sampling</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">y_pred</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">calc_best_radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">calc_best_cluster</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_predicted</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_centroid_sampling" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_combination_heatmap">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_combination_heatmap</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_combination_heatmap" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_consensus_threshold">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_consensus_threshold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">inds</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kmer_num</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_consensus_threshold" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_conservation">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_conservation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">coverage</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_ldict_ng</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fc_sm</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cns_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xsm</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_conservation" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots 3 charts depicting the kmer coverage, heatmap of coverage and a smoothed consensus by positon</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_coverage">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_conv</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radii</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_coverage" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_cutoffs">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_cutoffs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_r2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_cutoff_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'max'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_cutoffs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_feature_length">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_feature_length</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">features_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_feature_length" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_kmer_genome_distribution">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_kmer_genome_distribution</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">kmer_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_kmer_genome_distribution" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_probe_snps">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_probe_snps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">snp_dict</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_tmax</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_probe_snps" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_pssm">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_pssm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tdf</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_pssm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_score">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_scores</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">spacer_length</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_score" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_spacer_p_distribution">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_spacer_p_distribution</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_spacer_p_distribution" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_target_unp">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_target_unp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_unp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_len</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_target_unp" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots the local structure accessibility of a target genome. Used for vienna.</p>
<p>Inputs:
df_unp: output of vienna_plfold function that is a df
window_len: int - length of the window used to calculate local structure
title: str - name of the target</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_tree">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_tree</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tpath</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tree</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fig_scale</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_tree" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.plot.plot_unique_seqs">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.plot.</span></span><span class="sig-name descname"><span class="pre">plot_unique_seqs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_r2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.plot.plot_unique_seqs" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.presets">
<span id="dtrapipeline-utils-presets-module"></span><h2>DTRAPipeline.utils.presets module<a class="headerlink" href="#module-DTRAPipeline.utils.presets" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.BaseType">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">BaseType</span></span><a class="headerlink" href="#DTRAPipeline.utils.presets.BaseType" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>presets for inspectr, sherlock based on genome size.
current options are small, medium, large.</p>
<p>TODO:</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.BaseType.set_params">
<span class="sig-name descname"><span class="pre">set_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">param_dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.BaseType.set_params" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.BaseType.set_platform">
<span class="sig-name descname"><span class="pre">set_platform</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.BaseType.set_platform" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.EnzymeBase">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">EnzymeBase</span></span><a class="headerlink" href="#DTRAPipeline.utils.presets.EnzymeBase" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Large">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">Large</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.Large" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.BaseType" title="DTRAPipeline.utils.presets.BaseType"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.BaseType</span></code></a></p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Lwa13Base">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">Lwa13Base</span></span><a class="headerlink" href="#DTRAPipeline.utils.presets.Lwa13Base" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.EnzymeBase" title="DTRAPipeline.utils.presets.EnzymeBase"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.EnzymeBase</span></code></a></p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Medium">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">Medium</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.Medium" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.BaseType" title="DTRAPipeline.utils.presets.BaseType"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.BaseType</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Medium.set_platform">
<span class="sig-name descname"><span class="pre">set_platform</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.Medium.set_platform" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.RS9Base">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">RS9Base</span></span><a class="headerlink" href="#DTRAPipeline.utils.presets.RS9Base" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.EnzymeBase" title="DTRAPipeline.utils.presets.EnzymeBase"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.EnzymeBase</span></code></a></p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Small">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">Small</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.Small" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.BaseType" title="DTRAPipeline.utils.presets.BaseType"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.BaseType</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.Small.set_platform">
<span class="sig-name descname"><span class="pre">set_platform</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">platform</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.Small.set_platform" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.presets.get_presets">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.presets.</span></span><span class="sig-name descname"><span class="pre">get_presets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">set_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">platform</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.presets.get_presets" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
df_r2: filtered references dataframe from FastaParse object filtered_refs
set_type: str, optionally set genome size preferences from [small, medium, large]
platform: str, available options are [sherlock, SLK, inspectr, ISR]
get preset based on sequences</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.refeature">
<span id="dtrapipeline-utils-refeature-module"></span><h2>DTRAPipeline.utils.refeature module<a class="headerlink" href="#module-DTRAPipeline.utils.refeature" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.refeature.</span></span><span class="sig-name descname"><span class="pre">GenBankReFeature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">dict</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Module to reorient every feature in every genome as a pseudoalignment.
The reoriented files can then continue through the rest of the pipeline in a 
standard fashion.</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.get_base_file_name">
<span class="sig-name descname"><span class="pre">get_base_file_name</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.get_base_file_name" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.get_fasta_regions_df">
<span class="sig-name descname"><span class="pre">get_fasta_regions_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">features_filename</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.get_fasta_regions_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Executes a blast query for the genbank features and reorients the features of every fasta record in a dataframe</p>
<p>Input:
features_filename: str, features fasta file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.get_features_df">
<span class="sig-name descname"><span class="pre">get_features_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">lower_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">750</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upper_limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">inf</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">locus_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.get_features_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the feature finder, retrieves the representative fasta file, and writes to a fasta</p>
<p>Input:
lower_limit: int, lower limit for feature size
upper_limit: int, upper limit for feature size
gene_list: list, list of genes to focus on
locus_list: list, list of locus to focus on</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.microbe_process">
<span class="sig-name descname"><span class="pre">microbe_process</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gene_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">locus_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.microbe_process" title="Permalink to this definition">¶</a></dt>
<dd><p>Main Function for the ReFeatures operations which will reorient, reset, and reinput the
paramaters with the new fasta files for further analysis.</p>
<p>Input:
gene_list: list, list of genes to focus on
locus_list: list, list of locus to focus on</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.pull_marker">
<span class="sig-name descname"><span class="pre">pull_marker</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">record</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">blast_df</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.pull_marker" title="Permalink to this definition">¶</a></dt>
<dd><p>Pulls marker areas from the blast df file and formats them into a dataframe</p>
<p>Input:
record: Bio.Seq.Seq, biopython seq record
blast_df: pd.DataFrame, pandas dataframe of blast results</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.set_base">
<span class="sig-name descname"><span class="pre">set_base</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.set_base" title="Permalink to this definition">¶</a></dt>
<dd><p>Inherit parameters for input into supporting modules.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.set_blast_features">
<span class="sig-name descname"><span class="pre">set_blast_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">preset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.set_blast_features" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.refeature.GenBankReFeature.set_key_presets">
<span class="sig-name descname"><span class="pre">set_key_presets</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.refeature.GenBankReFeature.set_key_presets" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.seqformatter">
<span id="dtrapipeline-utils-seqformatter-module"></span><h2>DTRAPipeline.utils.seqformatter module<a class="headerlink" href="#module-DTRAPipeline.utils.seqformatter" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Alphabet">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">Alphabet</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Alphabet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Alphabet.all_letters">
<span class="sig-name descname"><span class="pre">all_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Alphabet.all_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Alphabet.extended_letters">
<span class="sig-name descname"><span class="pre">extended_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Alphabet.extended_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Alphabet.generate_random_sequence">
<span class="sig-name descname"><span class="pre">generate_random_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Alphabet.generate_random_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a random sequence drawn from an alphabet.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><p>length (int): length of random sequence to be generated
sequence_type (chr): one of the allowed ALPHABETS (nucleic_acid, protein, glycan)</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Alphabet.standard_letters">
<span class="sig-name descname"><span class="pre">standard_letters</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Alphabet.standard_letters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Error">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">Error</span></span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Error" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Exception</span></code></p>
<p>Base class for exceptions in this module.</p>
</dd></dl>

<dl class="py exception">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.InputError">
<em class="property"><span class="pre">exception</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">InputError</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">message</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.InputError" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.seqformatter.Error" title="DTRAPipeline.utils.seqformatter.Error"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.seqformatter.Error</span></code></a></p>
<p>Exception raised for errors in the input.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><p>expression: input expression in which the error occurred
message: explanation of the errors</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">Sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence.augmented">
<span class="sig-name descname"><span class="pre">augmented</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">augmentation_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence.augmented" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the augmented (complemented/reverse complemented) sequence. Applicable for nucleic_acid only!</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence.fill">
<span class="sig-name descname"><span class="pre">fill</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence.fill" title="Permalink to this definition">¶</a></dt>
<dd><p>Replace gaps (“-“) with the corresponding gap letter.
Not done for glycans.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence.pad">
<span class="sig-name descname"><span class="pre">pad</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence.pad" title="Permalink to this definition">¶</a></dt>
<dd><p>Pad sequence to desired length using the appropriate gap letter</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence.scrambled">
<span class="sig-name descname"><span class="pre">scrambled</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence.scrambled" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the scrambled sequence</p>
<p>Ensures that the scrambled output returned has the same letter frequency
but in a different order than the original sequence. Negative control.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.Sequence.standardized">
<span class="sig-name descname"><span class="pre">standardized</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.Sequence.standardized" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the fully standardized sequence</p>
<p>Fill gaps, add padding, map non-standard letters and ensure all letters are valid</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.annotate_contig">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">annotate_contig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.annotate_contig" title="Permalink to this definition">¶</a></dt>
<dd><p>This function locates a contig within a reference genome and assigns an indicator
depending on how many instances that contig is found within the reference genome.</p>
<p>Inputs:
seq: str - nucleic acid
ref_genome: str - nucleic acid</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.clean_one_sequence">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">clean_one_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nucleic_acid'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ret_scrambled</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.clean_one_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the datacleaner class to clean a sequence and return a scrambled
copy if desired.</p>
<p>Inputs:
seq: str
sequence_type: can be ‘nucleic_acid’ or ‘protein’
ret_scrambled: Bool</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.compute_contig_coords">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">compute_contig_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cleaned_contigs</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.compute_contig_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>This function identifies the positions of the cleaned contigs relative to a reference genome.</p>
<p>Inputs:
cleaned_contigs: str - list of contigs.</p>
<p>ref_genome: str - sequence of the ref genome that contigs belong to</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.edlib_pw_align">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">edlib_pw_align</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.edlib_pw_align" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.format_header">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">format_header</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta_header</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.format_header" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.process_glycans">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">process_glycans</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">glycan_list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.process_glycans" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper function to process list of glycans into glycoletters.
Input is a list of glycans. Output is a list of glycowords</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="DTRAPipeline.utils.seqformatter.small_glymotif_find">
<span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.seqformatter.</span></span><span class="sig-name descname"><span class="pre">small_glymotif_find</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">s</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.seqformatter.small_glymotif_find" title="Permalink to this definition">¶</a></dt>
<dd><p>Breaks down glycans into list of monomers. Assumes
that user provides glycans in a consistent universal
notation. Input is a glycan string</p>
</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.taxonomy">
<span id="dtrapipeline-utils-taxonomy-module"></span><h2>DTRAPipeline.utils.taxonomy module<a class="headerlink" href="#module-DTRAPipeline.utils.taxonomy" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.taxonomy.</span></span><span class="sig-name descname"><span class="pre">Assembly</span></span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Downloads entire taxonomy structure, compatible with genbank and refseq.</p>
<p>creates a bash file which automaticall downloads all selected genomes. Bash file
created to keep a log of all genomes contained within folder.</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.download_genomes">
<span class="sig-name descname"><span class="pre">download_genomes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'raw_fna_files/'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.download_genomes" title="Permalink to this definition">¶</a></dt>
<dd><p>Calls get_assembly_ftp_bash and executes.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.get_assembly">
<span class="sig-name descname"><span class="pre">get_assembly</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">option</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genbank'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">force</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.get_assembly" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
option: str, [genbank, refseq]
force: bool</p>
<p>Dowloads summary text files from ncbi.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.get_assembly_ftp_bash">
<span class="sig-name descname"><span class="pre">get_assembly_ftp_bash</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'get_genomes.sh'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.get_assembly_ftp_bash" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates bash file to download selected genomes.</p>
<p>Input:
file_name: str, name of file
out_dir: str, path to output directory</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.get_rep_genomes">
<span class="sig-name descname"><span class="pre">get_rep_genomes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'representatives_fna_files'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.get_rep_genomes" title="Permalink to this definition">¶</a></dt>
<dd><p>Inputs: kwargs
out_dir: str, location to save representatives</p>
<p>Retrieves all reference genomes for set taxids</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.load_assembly_df">
<span class="sig-name descname"><span class="pre">load_assembly_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">option</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genbank'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.load_assembly_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Inputs: kwargs, optional
path: str, path to assembly file
option: str, [genbank, refseq]</p>
<p>Load assembly table from file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Assembly.set_assembly_taxids">
<span class="sig-name descname"><span class="pre">set_assembly_taxids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Assembly.set_assembly_taxids" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
taxids = leaf_taxids</p>
<p>Set assembly taxids to download</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Taxa">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.taxonomy.</span></span><span class="sig-name descname"><span class="pre">Taxa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">nodes_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Taxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Optional kwarg:
nodes_path: str
names_path: str</p>
<p>Taxonomy structure to pull and download ncbi taxonomy files to determine full lineage upstream
and downstream from any particular taxid(s). Can also retrieve taxids from exact match names.
Phylogeny dataframes can be retrieved.</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Taxa.find_child_nodes">
<span class="sig-name descname"><span class="pre">find_child_nodes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Taxa.find_child_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Input: list of taxids</p>
<p>Finds all leaf nodes for every taxid in taxids including taxid.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Taxa.find_nodes">
<span class="sig-name descname"><span class="pre">find_nodes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Taxa.find_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>”
Input: list of taxids</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Taxa.get_phylo_df">
<span class="sig-name descname"><span class="pre">get_phylo_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Taxa.get_phylo_df" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.taxonomy.Taxa.get_taxid_from_name">
<span class="sig-name descname"><span class="pre">get_taxid_from_name</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.taxonomy.Taxa.get_taxid_from_name" title="Permalink to this definition">¶</a></dt>
<dd><p>Input: exact taxonomy name, no homotypic synonyms.</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.uclustutils">
<span id="dtrapipeline-utils-uclustutils-module"></span><h2>DTRAPipeline.utils.uclustutils module<a class="headerlink" href="#module-DTRAPipeline.utils.uclustutils" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.uclustutils.</span></span><span class="sig-name descname"><span class="pre">Centroid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_cluster_num</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">200</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Class to automatically calculate the range of centroids (representative cluster sequences)
for further use in the pipeline.</p>
<p>Is additionally capable of fitting the clustering rate to a gompertz model.</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.calculate_cluster_coverage">
<span class="sig-name descname"><span class="pre">calculate_cluster_coverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fdict</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.calculate_cluster_coverage" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.calculate_radius">
<span class="sig-name descname"><span class="pre">calculate_radius</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.calculate_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.exponential">
<span class="sig-name descname"><span class="pre">exponential</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.exponential" title="Permalink to this definition">¶</a></dt>
<dd><p>alternative exponential equation which has difficulty modeling the curve at saturation.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.get_optimal_stats">
<span class="sig-name descname"><span class="pre">get_optimal_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fdict</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.get_optimal_stats" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.get_representative">
<span class="sig-name descname"><span class="pre">get_representative</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rep_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'centroid_representative.fasta'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_seq</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.95</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.get_representative" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.gompertz_compute_radii">
<span class="sig-name descname"><span class="pre">gompertz_compute_radii</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">y</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.gompertz_compute_radii" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.modified_gompertz">
<span class="sig-name descname"><span class="pre">modified_gompertz</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xi</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.modified_gompertz" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates f(X) where f is the Modified Gompertz equation and X is inputs</p>
<p>Inputs:
x: float - radii
Nf: float - max seqs
b: float - initial signal growth rate
xi: float - radii shift</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.optimize_curve">
<span class="sig-name descname"><span class="pre">optimize_curve</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.optimize_curve" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.parse_centroid">
<span class="sig-name descname"><span class="pre">parse_centroid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">radius</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centroid</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fdict</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_out</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">40</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.parse_centroid" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.run">
<span class="sig-name descname"><span class="pre">run</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">step</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.run" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.sample_radius">
<span class="sig-name descname"><span class="pre">sample_radius</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.sample_radius" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.sum_above">
<em class="property"><span class="pre">static</span> </em><span class="sig-name descname"><span class="pre">sum_above</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.sum_above" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.Centroid.write_combine_clusters">
<span class="sig-name descname"><span class="pre">write_combine_clusters</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_conv</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.Centroid.write_combine_clusters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.UCLUSTWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.uclustutils.</span></span><span class="sig-name descname"><span class="pre">UCLUSTWrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.UCLUSTWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.uclustutils.UCLUSTWrapper.cluster_fasta">
<span class="sig-name descname"><span class="pre">cluster_fasta</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centroid_outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_full_cluster</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.uclustutils.UCLUSTWrapper.cluster_fasta" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.vienna">
<span id="dtrapipeline-utils-vienna-module"></span><h2>DTRAPipeline.utils.vienna module<a class="headerlink" href="#module-DTRAPipeline.utils.vienna" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.vienna.</span></span><span class="sig-name descname"><span class="pre">ViennaUtils</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">parser</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><a class="reference internal" href="DTRAPipeline.core.html#DTRAPipeline.core.parser.FastaParse" title="DTRAPipeline.core.parser.FastaParse"><span class="pre">DTRAPipeline.core.parser.FastaParse</span></a></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#DTRAPipeline.utils.presets.EnzymeBase" title="DTRAPipeline.utils.presets.EnzymeBase"><code class="xref py py-class docutils literal notranslate"><span class="pre">DTRAPipeline.utils.presets.EnzymeBase</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.add_duplex_cofold_vals">
<span class="sig-name descname"><span class="pre">add_duplex_cofold_vals</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">primary_df</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.add_duplex_cofold_vals" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.annotate_seq_uniqueness">
<span class="sig-name descname"><span class="pre">annotate_seq_uniqueness</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">spacer_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orient</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'minus'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.annotate_seq_uniqueness" title="Permalink to this definition">¶</a></dt>
<dd><p>This function locates a spacer within a target genome and assigns an indicator
depending on how many copies of that spacer are found.</p>
<p>Inputs:
spacer_seq: str - nucleic acid sequence of the spacer region
target_seq: str - nucleic acid sequence of the target
orient: str - orientation of the spacer which is either the rc (minus) or sense (plus)</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.calc_biopython_tm">
<span class="sig-name descname"><span class="pre">calc_biopython_tm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dntp_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dna_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">112.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.calc_biopython_tm" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the biopython module to esimate the Tm of a RNA-DNA hybrid sequence</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_heterodimer">
<span class="sig-name descname"><span class="pre">calc_primer3_heterodimer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seq2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temp_c</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">60</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dntp_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dna_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">112.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_loop</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_structure</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_heterodimer" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the primer3 bindings to calculate the heterodimerization thermodynamics of two sequences</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_homodimer">
<span class="sig-name descname"><span class="pre">calc_primer3_homodimer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">temp_c</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">60</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dntp_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dna_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">112.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_loop</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_structure</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_homodimer" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the primer3 bindings to calculate the homodimerization thermodynamics of a sequence.</p>
<p>Inputs:</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_tm">
<span class="sig-name descname"><span class="pre">calc_primer3_tm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dntp_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dna_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">112.5</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.calc_primer3_tm" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the primer3 module to esimate the Tm of a nucleic_acid sequence</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.calculate_crRNA_optimized_guides">
<span class="sig-name descname"><span class="pre">calculate_crRNA_optimized_guides</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filio</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.calculate_crRNA_optimized_guides" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.compute_RNAsTar_interaction">
<span class="sig-name descname"><span class="pre">compute_RNAsTar_interaction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">target</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_name</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.compute_RNAsTar_interaction" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the interaction of a set of probes against a target sequence
using the IntaRNAsTar mode. Warning - will result in truncations in Nqueries &gt; 500.</p>
<p>Inputs:
target: fasta file containing target genome
query: fasta file containing the probe seqs
out_name: str - name of the output file
temperature: float which by default is 22C</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.compute_punp_probe">
<span class="sig-name descname"><span class="pre">compute_punp_probe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_unp</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orient</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'plus'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.compute_punp_probe" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts the average probability that probe is unpaired in the target</p>
<p>Inputs:
probe: str - nucleic acid sequence of the probe
target: str - nucleic acid sequence of the target
df_unp: df that is the output of RNAplfold
orient: orientation of the probe relative to the target, can be ‘minus’ or ‘plus’</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.escape_ansi">
<em class="property"><span class="pre">static</span> </em><span class="sig-name descname"><span class="pre">escape_ansi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">line</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.escape_ansi" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove ANSI color coded command line characters from string</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.evaluate_primer3_hairpin">
<span class="sig-name descname"><span class="pre">evaluate_primer3_hairpin</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">temp_c</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">60</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dv_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">8.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dntp_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dna_conc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">112.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_loop</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.evaluate_primer3_hairpin" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.extract_title">
<em class="property"><span class="pre">static</span> </em><span class="sig-name descname"><span class="pre">extract_title</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">title</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.extract_title" title="Permalink to this definition">¶</a></dt>
<dd><p>pull out the gene location, feature type, feature name, accession from the fasta header string as
formatted using the MSABase.guide_features_annotate method</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.generate_random_sequence">
<span class="sig-name descname"><span class="pre">generate_random_sequence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">length</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.generate_random_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a random sequence drawn from an alphabet.</p>
<p>Inputs:
length (int): length of random sequence to be generated</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.get_base_file_name">
<span class="sig-name descname"><span class="pre">get_base_file_name</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.get_base_file_name" title="Permalink to this definition">¶</a></dt>
<dd><p>retrieve parser base file name</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.get_rep_path">
<span class="sig-name descname"><span class="pre">get_rep_path</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.get_rep_path" title="Permalink to this definition">¶</a></dt>
<dd><p>retrieve the parser representative path</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.get_rfold">
<span class="sig-name descname"><span class="pre">get_rfold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fileio</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.get_rfold" title="Permalink to this definition">¶</a></dt>
<dd><p>Process guides through vienna’s RNAfold module
fileio: filehandle, handle for fasta</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.get_spacer_neighborhood">
<span class="sig-name descname"><span class="pre">get_spacer_neighborhood</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">spacer_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">orient</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'plus'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.get_spacer_neighborhood" title="Permalink to this definition">¶</a></dt>
<dd><p>This function gets the corresponding target neighborhood around the spacer sequence.</p>
<p>Inputs:
spacer_seq: str - nucleic_acid sequence of the spacer
target_seq: str - nucleic_acid sequence of the target
spacer_neighborhood: int - size of the neighborhood in the target around the spacer, with the spacer at the center
orient: str - orientation of the spacer which is either the rc (minus) or sense (plus)</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.make_basic_df">
<span class="sig-name descname"><span class="pre">make_basic_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">header_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.make_basic_df" title="Permalink to this definition">¶</a></dt>
<dd><p>probe_list: list, list of nucleotide strings
header_list: list, list of int, str, float</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.make_primary_df">
<span class="sig-name descname"><span class="pre">make_primary_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">probe_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">representative_fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_unp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compute_punp_probe</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.make_primary_df" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.mutate_seq">
<span class="sig-name descname"><span class="pre">mutate_seq</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mm_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_random</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">loc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.mutate_seq" title="Permalink to this definition">¶</a></dt>
<dd><p>This function introduces mutations into a sequence and enumerates all possible unique mutants. Mutation
position can be chosen at random or deliberately introduced.</p>
<p>Inputs:
seq: str - nucleic_acid sequence
mm_size: int - number of consecutive mismatches to introduce in nt.
do_random: bool - if True, will randomly choose mutation positon
loc: can be None or int but has to be within the domain of seq</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.parse_cofold_output">
<span class="sig-name descname"><span class="pre">parse_cofold_output</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cofold</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.parse_cofold_output" title="Permalink to this definition">¶</a></dt>
<dd><p>This function parses the output of vienna_cofold which is a list of dictionaries.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.parse_vienna_duplex_out">
<span class="sig-name descname"><span class="pre">parse_vienna_duplex_out</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.parse_vienna_duplex_out" title="Permalink to this definition">¶</a></dt>
<dd><p>Parses the output of Vienna RNADuplex to extract the
start and end positions in the target where the probe is predicted to hybridize
as well as its energetics.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.pick_top_vienna_guides">
<span class="sig-name descname"><span class="pre">pick_top_vienna_guides</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.pick_top_vienna_guides" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to run guides through vienna and pick the best guides based per gene.
Continues to pull guides until it it retrieves sufficent guides.</p>
<p>df: feature_df, generated from the features module</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.run">
<span class="sig-name descname"><span class="pre">run</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">eua</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">run_vienna</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.run" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.score_seq_complexity">
<span class="sig-name descname"><span class="pre">score_seq_complexity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.score_seq_complexity" title="Permalink to this definition">¶</a></dt>
<dd><p>This function scores the complexity (normalized entropy) of a sequence and normalizes between 0 and 1.
More random sequences are weighted towards 1 whereas homopolymers are weighted towards 0.</p>
<p>Inputs:
seq - str: nucleic_acid sequence</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.set_enzyme_presets">
<span class="sig-name descname"><span class="pre">set_enzyme_presets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">enzyme_preset</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.set_enzyme_presets" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the enzyme base from string. Current options are RS9 and Lwa13</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.vienna_RNAduplex">
<span class="sig-name descname"><span class="pre">vienna_RNAduplex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">spacer</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rep_seq</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.vienna_RNAduplex" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the ViennaRNA RNAduplex package to calculate the folding metrics for two RNA sequences.
Returns the mfe of the duplex.</p>
<p>Inputs:
s1: str - nucleic-acid sequence of a probe
s2: str - nucleic-acid sequence of a long target genome
temperature: float - in celsius</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.vienna_cofold">
<span class="sig-name descname"><span class="pre">vienna_cofold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">guide_conc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_conc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_dp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.vienna_cofold" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the ViennaRNA co-fold package to calculate the folding metrics for two RNA sequences.
A coarse pair-probability plot is computed by RNAcofold and saved as dot.ps in the current
working directory</p>
<p>Inputs:
s1: str - nucleic-acid sequence of a probe
s2: str - nucleic-acid sequence of a target
s1_conc: float - concentration of s1 in M
s2_conc: float - concentration of s2 in M
temperature: float - in celsius
return_dp: bool - if True will parse the .ps file and return a square np array of</p>
<blockquote>
<div><p>pair probabilities. This is different from NuPack because there’s no
concentration correction and salt correction.</p>
</div></blockquote>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.vienna.ViennaUtils.vienna_plfold">
<span class="sig-name descname"><span class="pre">vienna_plfold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">return_dp</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.vienna.ViennaUtils.vienna_plfold" title="Permalink to this definition">¶</a></dt>
<dd><p>This function uses the RNAplfold routine to locally solve the structure of a long genomic target
Warning! Uses the default settings for RNAplfold</p>
<p>Inputs:
target_path: path to a fasta file for the target genome
target_name: str - name of the target
window_len: int - window size in nt over which to locally compute structure
guide_size: int - hybridization length of the probe with the target genome.
temperature: float - in celsius
return_dp: bool - if True will parse the .ps file and return a square np array of</p>
<blockquote>
<div><p>pair probabilities. This is different from NuPack because there’s no
concentration correction and salt correction.</p>
</div></blockquote>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils.wrappers">
<span id="dtrapipeline-utils-wrappers-module"></span><h2>DTRAPipeline.utils.wrappers module<a class="headerlink" href="#module-DTRAPipeline.utils.wrappers" title="Permalink to this headline">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.BOWTIE2Wrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">BOWTIE2Wrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.BOWTIE2Wrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.BOWTIE2Wrapper.build">
<span class="sig-name descname"><span class="pre">build</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_file_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.BOWTIE2Wrapper.build" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">EntrezWrapper</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">email</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncbi_api_key</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Downloads files from the NCBI nucleotide database.</p>
<p>Input:
email: str, entrez requires an email for every query
ncbi_api_key: str, strongly reccomended to include this for large queries</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.format_assembly_params">
<span class="sig-name descname"><span class="pre">format_assembly_params</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assembly_db</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_reps</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">q_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.format_assembly_params" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_assembly_fastas">
<span class="sig-name descname"><span class="pre">get_assembly_fastas</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_reps</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assembly_db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genbank'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">q_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genome'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">read_only</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">individual</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">merge_contigs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">retmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_assembly_fastas" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to retrieve fastas from the ncbi assembly database.</p>
<p>Input:
query: str, ncbi search query
out_dir: str, path to output directory
only_reps: bool, find reference or representative genomes
assembly_db: str, genbank or refseq
q_type: str, possible options are genome, partial, or custom. Defaults will assist in proper genome retrieval
read_only: bool, returns the count and search results if they exist without downloading</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_assembly_gbff">
<span class="sig-name descname"><span class="pre">get_assembly_gbff</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">acc_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">limit</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_assembly_gbff" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_elink_dict">
<span class="sig-name descname"><span class="pre">get_elink_dict</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">id_term</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nuccore'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'assembly'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_elink_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve the elink associated with a accession number to retrieve the corresponding
assembly entry. Allows the program to use contig and scaffolding accession ids if
no master accession id can be found.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_gbff_handle">
<span class="sig-name descname"><span class="pre">get_gbff_handle</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">acc_id</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">write</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_gbff_handle" title="Permalink to this definition">¶</a></dt>
<dd><p>Get nucleotide gbff file</p>
<p>Input:
acc_id: str, ncbi accession id from fasta header.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_nucleotide_fasta">
<span class="sig-name descname"><span class="pre">get_nucleotide_fasta</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">q_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genome'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nucleotide'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">write</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">retmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_nucleotide_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>”
Input:
query: str, ncbi nucleotide search term, should be pathogen
outfile: str, path to output fasta file
q_type: str, [genome, plasmid, custom]</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_search_dict">
<span class="sig-name descname"><span class="pre">get_search_dict</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">database</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mquery</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sort</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'significance'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_search_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.get_taxa_lineage">
<span class="sig-name descname"><span class="pre">get_taxa_lineage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.get_taxa_lineage" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve the full taxonomic lineage</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.parse_taxa_result">
<span class="sig-name descname"><span class="pre">parse_taxa_result</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tdict</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.parse_taxa_result" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.EntrezWrapper.write_merged_assembly">
<span class="sig-name descname"><span class="pre">write_merged_assembly</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bstr</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">only_reps</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.EntrezWrapper.write_merged_assembly" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.GetTaxids">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">GetTaxids</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.GetTaxids" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.GetTaxids.by_name">
<span class="sig-name descname"><span class="pre">by_name</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.GetTaxids.by_name" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a taxid from a given ncbi accepted name</p>
<p>name: str, taxa name</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.GetTaxids.by_taxid">
<span class="sig-name descname"><span class="pre">by_taxid</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxid</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.GetTaxids.by_taxid" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves a list of phylogenetic children taxids from given taxid</p>
<p>taxid: int, taxid identifier</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MAFFTWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">MAFFTWrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MAFFTWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MAFFTWrapper.auto">
<span class="sig-name descname"><span class="pre">auto</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap_penalty</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extension_penalty</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reorder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MAFFTWrapper.auto" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
fasta_file: str, path to fasta file for MAFFT alignment
output_file: str, path to aligned output fasta file.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MAFFTWrapper.auto_ref">
<span class="sig-name descname"><span class="pre">auto_ref</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gap_penalty</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extension_penalty</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reorder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MAFFTWrapper.auto_ref" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
fasta_file: str, path to fasta file for MAFFT alignment
ref_file: str, path to reference fasta file to align all other fastas to.
output_file: str, path to aligned output fasta file.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MashWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">MashWrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MashWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MashWrapper.dist">
<span class="sig-name descname"><span class="pre">dist</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reference_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subject_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">48</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MashWrapper.dist" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MashWrapper.info">
<span class="sig-name descname"><span class="pre">info</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hash_file</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MashWrapper.info" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.MashWrapper.sketch">
<span class="sig-name descname"><span class="pre">sketch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">reference_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">48</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kmer_copy_num</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.MashWrapper.sketch" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">NCBIBlastn</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_isr_specificity">
<span class="sig-name descname"><span class="pre">blastn_isr_specificity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative_taxidlist_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nt'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_isr_specificity" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_remote">
<span class="sig-name descname"><span class="pre">blastn_remote</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cmd</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'blastn'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nt'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">task</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'blastn-short'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">perc_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qcov_hsp_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_target_seqs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">50</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_remote" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
query: str, path to fasta file of universal regions
out: str, path to output blast tab file
perc_identity: int, percentage identical to query entries
qcov_hsp_perc: int, percentage of query entry covered by hits
max_target_seqs: int, number of hits to show in tab file</p>
<p>Used for specificity analysis of final universal regions.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_specificity">
<span class="sig-name descname"><span class="pre">blastn_specificity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">negative_taxidlist_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'nt'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">task</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'megablast'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">perc_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qcov_hsp_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_target_seqs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">15</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_specificity" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
query: str, path to fasta file of universal regions
out: str, path to output blast tab file
perc_identity: int, percentage identical to query entries
qcov_hsp_perc: int, percentage of query entry covered by hits
max_target_seqs: int, number of hits to show in tab file</p>
<p>Used for specificity analysis of final universal regions.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_subject">
<span class="sig-name descname"><span class="pre">blastn_subject</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subject</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">task</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'blastn-short'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">perc_identity</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qcov_hsp_perc</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">92</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_target_seqs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buffer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">word_size</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn.blastn_subject" title="Permalink to this definition">¶</a></dt>
<dd><p>Input:
query: str, path to fasta file of universal regions
subject: str, path to curated fasta file list
out: str, path to output blast tab file
perc_identity: int, percentage identical to query entries
qcov_hsp_perc: int, percentage of query entry covered by hits
max_target_seqs: int, number of hits to show in tab file</p>
<p>Used for sensitivity analysis of final universal regions.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.NCBIBlastn.fetch_taxdb">
<span class="sig-name descname"><span class="pre">fetch_taxdb</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.NCBIBlastn.fetch_taxdb" title="Permalink to this definition">¶</a></dt>
<dd><p>This file is necessary to run specificity blast. A copy must be held locally.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.ProgressiveMauveWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">ProgressiveMauveWrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.ProgressiveMauveWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.ProgressiveMauveWrapper.align">
<span class="sig-name descname"><span class="pre">align</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fasta_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_guide_tree</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">backbone_output</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_scaled_penalty</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">5000</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.ProgressiveMauveWrapper.align" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">DTRAPipeline.utils.wrappers.</span></span><span class="sig-name descname"><span class="pre">SourMashWrapper</span></span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Wrapper for sourmash</p>
<dl class="simple">
<dt>public methods:</dt><dd><p>compare
compute
search
signature_cat
signature_subtract
index</p>
</dd>
</dl>
<p>database types: virus, prok, fungi, human</p>
<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.compare">
<span class="sig-name descname"><span class="pre">compare</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signatures</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">csv_out</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.compare" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.compute">
<span class="sig-name descname"><span class="pre">compute</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">seqfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sour_scaled</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">merged_sig</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stderr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stdout</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.compute" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.index">
<span class="sig-name descname"><span class="pre">index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">db_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signature</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stderr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stdout</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.index" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.search">
<span class="sig-name descname"><span class="pre">search</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">guide_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">signature</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">database_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outfile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1e-05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stderr</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">stdout</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.search" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.signature_cat">
<span class="sig-name descname"><span class="pre">signature_cat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">signature_directory</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">outfile</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.signature_cat" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="DTRAPipeline.utils.wrappers.SourMashWrapper.signature_subtract">
<span class="sig-name descname"><span class="pre">signature_subtract</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">main_sig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">subtract_sig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_sig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">guide_size</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#DTRAPipeline.utils.wrappers.SourMashWrapper.signature_subtract" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-DTRAPipeline.utils">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-DTRAPipeline.utils" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">DTRAPipe</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="DTRAPipeline.html">DTRAPipeline package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="DTRAPipeline.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="DTRAPipeline.core.html">DTRAPipeline.core package</a></li>
<li class="toctree-l3"><a class="reference internal" href="DTRAPipeline.databases.html">DTRAPipeline.databases package</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">DTRAPipeline.utils package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="DTRAPipeline.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="DTRAPipeline.html#module-DTRAPipeline.cmd">DTRAPipeline.cmd module</a></li>
<li class="toctree-l2"><a class="reference internal" href="DTRAPipeline.html#module-DTRAPipeline">Module contents</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="DTRAPipeline.html">DTRAPipeline package</a><ul>
      <li>Previous: <a href="DTRAPipeline.databases.html" title="previous chapter">DTRAPipeline.databases package</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, AWL, PR.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.2.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/DTRAPipeline.utils.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>